### PR TITLE
fix memory issue dti metrics

### PR DESCRIPTION
**Pull request by arnaudbore**

### CODE REVIEW

**Review by mdesco — APPROVED** [review]
This works well!

File: scripts/scil_compute_dti_metrics.py [hunk 1]
```diff
@@ -255,38 +269,40 @@ def main():
...
     if args.evecs:
         evecs = tenfit.evecs.astype(np.float32)
         evecs_img = nib.Nifti1Image(evecs, affine)
         nib.save(evecs_img, args.evecs)
 
         # save individual e-vectors also
-        e1_img = nib.Nifti1Image(evecs[..., 0], affine)
-        e2_img = nib.Nifti1Image(evecs[..., 1], affine)
-        e3_img = nib.Nifti1Image(evecs[..., 2], affine)
+        for i in [1, 2, 3]:
```
Comment (GuillaumeTh) [inline scripts/scil_compute_dti_metrics.py:287]: range(3)

Comment (GuillaumeTh) [reply scripts/scil_compute_dti_metrics.py:287]: After using range adapt below.

File: scripts/scil_compute_dti_metrics.py [hunk 2]
```diff
@@ -255,38 +269,40 @@ def main():
...
     if args.evals:
         evals = tenfit.evals.astype(np.float32)
         evals_img = nib.Nifti1Image(evals, affine)
         nib.save(evals_img, args.evals)
 
         # save individual e-values also
-        e1_img = nib.Nifti1Image(evals[..., 0], affine)
-        e2_img = nib.Nifti1Image(evals[..., 1], affine)
-        e3_img = nib.Nifti1Image(evals[..., 2], affine)
+        for i in [1, 2, 3]:
```
Comment (GuillaumeTh) [inline scripts/scil_compute_dti_metrics.py:300]: Same comment use range please

File: scripts/scil_compute_dti_metrics.py [hunk 3]
```diff
@@ -324,19 +342,33 @@ def main():
...
+            if args.mask is not None:
+                tenfit2 = tenmodel.fit(data[i, :, :, :], mask[i, :, :])
+            else:
+                tenfit2 = tenmodel.fit(data[i, :, :, :])
+
+            data_diff[i, :, :, :] = np.abs(
+                tenfit2.predict(gtab, S0[i, :, :]).astype(
+                    np.float32) - data[i, :, :])
+
+        R = np.mean(data_diff, axis=-1)
```
Comment (GuillaumeTh) [inline scripts/scil_compute_dti_metrics.py:362]: np.mean(data_diff[..., ~gtab.b0s_mask], axis=-1, dtype=np.float32)

**Review by GuillaumeTh — CHANGES_REQUESTED** [review]
I will test after the fix

**Review by GuillaumeTh — APPROVED** [review]
